listenbrainz/webserver/static/js/src/utils/types.d.ts

```diff
@@ -137,7 +137,8 @@ declare type SpotifyPlayerSDKState = {
   position: number;
   duration: number;
   track_window: {
-    current_track: SpotifyTrack;
+    current_track: SpotifyTrack | null;
+    previous_tracks?: SpotifyTrack[]; // always an array even if empty
```

Comment:
Where and how is previous_tracks? I see a comment about it in SpotifyPlayer.tsx but there is no code change accompanying so a bit confused.

Reply:
Right, that's left over from me trying to fix the track end detection, but that was not compatible with another fix in this PR reported by mayhem (https://github.com/metabrainz/listenbrainz-server/pull/2054/files#diff-939dbd0e74ca9fa5dea63f2a3c6a7600a5de30a317cef08fd44f9dbb495f2712R471-R475)

I'll remove it